pkg/instancetype/instancetype.go

```diff
@@ -746,7 +746,7 @@ func (m *InstancetypeMethods) inferDefaultsFromDataVolume(vm *virtv1.VirtualMach
 			if dvt.Name != dvName {
 				continue
 			}
-			return m.inferDefaultsFromDataVolumeSpec(&dvt.Spec, defaultNameLabel, defaultKindLabel)
+			return m.inferDefaultsFromDataVolumeSpec(&dvt.Spec, defaultNameLabel, defaultKindLabel, vm.Namespace)
```

Comment:
@0xFelix did ask about this offline, I don't believe it can be when called from the mutation webhook, AFAIK the namespace is already populated by kube-apiserver before it calls down to our webhooks/.

Reply:
I did not check this case (I mean where is this function called) but it can be empty on the create.

Reply:
Yeah it can be in the definition but by the time we see it in our webhooks I believe it's always filled in. For example if I instrument the same area and apply without a namespace I see `default` used:

```
git diff
diff --git a/pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go b/pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go
index ed7c04735..b7a2db605 100644
--- a/pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go
+++ b/pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go
@@ -62,6 +62,8 @@ func (mutator *VMsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.
                return webhookutils.ToAdmissionResponseError(err)
        }
 
+       log.Log.Info(fmt.Sprintf("lyarwood - vm %s has namespace %s", vm.Name, vm.Namespace))
+
        // Validate the InstancetypeMatcher before proceeding, the schema check above isn't enough
yq .metadata.namespace,.namespace examples/vm-cirros.yaml
null
null
./cluster-up/kubectl.sh apply -f examples/vm-cirros.yaml
[..]
./cluster-up/kubectl.sh logs -n kubevirt virt-api-6ffdcd9fb7-psv99 | grep lyarwood
selecting docker as container runtime
{"component":"virt-api","level":"info","msg":"lyarwood - vm vm-cirros has namespace default","pos":"vm-mutator.go:65","timestamp":"2023-03-16T16:47:35.516484Z"}
{"component":"virt-api","level":"info","msg":"lyarwood - vm vm-cirros has namespace default","pos":"vm-mutator.go:65","timestamp":"2023-03-16T16:47:35.528289Z"}
{"component":"virt-api","level":"info","msg":"lyarwood - vm vm-cirros has namespace default","pos":"vm-mutator.go:65","timestamp":"2023-03-16T16:47:35.533902Z"}
{"component":"virt-api","level":"info","msg":"lyarwood - vm vm-cirros has namespace default","pos":"vm-mutator.go:65","timestamp":"2023-03-16T16:47:35.537931Z"}
{"component":"virt-api","level":"info","msg":"lyarwood - vm vm-cirros has namespace default","pos":"vm-mutator.go:65","timestamp":"2023-03-16T16:47:35.546099Z"}
```

Hopefully this isn't `kubectl` doing this, really hard to find docs on this unfortunatley.

Reply:
@xpivarc, appreciate the review, can you take another look if you have time as I'd like to get these closed out. Let me know if I can help with anything in return.